scripts/getting_started.sh

```diff
@@ -0,0 +1,583 @@
+
+grep ${KIND_IP} ${manageClusterConfFile} | grep -q 6443
+if [ $? -ne 0 ]; then
+    sed -i 's/    server\:.*/    server\: https\:\/\/'"${KIND_IP}"'\:6443/g' ${manageClusterConfFile}
+else
+    echo "Already modified ${manageClusterConfFile} before"
+fi
+
+
+## Register BYOH host to management cluster
+for i in {1..2}
+do
+  #Check if container "host$i" is already created.
+  docker ps -a | grep -q "host${i}"
+  if [ $? -ne 0 ]; then
+    runCmd "docker run --detach --tty --hostname host${i} --name host${i} --privileged --security-opt seccomp=unconfined --tmpfs /tmp --tmpfs /run --volume /var --volume /lib/modules:/lib/modules:ro --network kind byoh-dev/node:v1.22.3" 0 "Starting byoh container: host${i}..."
+    runByohAgent ${i} ${byohBinaryFile} ${manageClusterConfFile}
+  else
+    echo "Container \"host${i}\" is already created"
+
+    # check if docker status is valid
+    status=`docker container inspect -f '{{.State.Status}}' host${i}`
+    if [ "${status}" != "running" ]; then
+        echo "Error: Status of Container \"host${i}\" is ${status}, suggest to remove it, then rerun this script"
+        exit
+    fi
+
+    # check if byoh process running in this container
+    docker exec host${i} sh -c "ps aux" | grep -q byoh-hostagent
+    if [ $? -ne 0 ]; then
+        runByohAgent ${i} ${byohBinaryFile} ${manageClusterConfFile}
+    else
+        echo "byoh process is already ran in this container"
+    fi
+  fi
+done
+
+# Find a available IP for control plane endpoint
+CONTROL_PLANE_ENDPOINT_IP=""
+calcControlPlaneIP
+if [ -z "${CONTROL_PLANE_ENDPOINT_IP}" ]; then
+    echo "Can't get an available IP for control plane endpoint, exit...."
+    exit
+fi
+echo "CONTROL_PLANE_ENDPOINT_IP is ${CONTROL_PLANE_ENDPOINT_IP}"
+
+clusterYamlFile="/tmp/cluster-yaml"
+CONTROL_PLANE_ENDPOINT_IP=${CONTROL_PLANE_ENDPOINT_IP} clusterctl generate cluster byoh-cluster --infrastructure byoh --kubernetes-version v1.22.3 --control-plane-machine-count 1  --worker-machine-count 1 --flavor docker > "${clusterYamlFile}"
+if [ $? -ne 0 ]; then
+    echo "Generate ${clusterYamlFile} failed, exiting..."
+    exit
+fi
+
+
+echo "Creating the workload cluster..."
+kubectl apply -f ${clusterYamlFile} 
+if [ $? -ne 0 ]; then
+    echo "Create the workload cluster failed"
+    exit
+fi
+
+echo "Retrieving the kubeconfig of workload cluster..."
+
+maxRunTimes=10
+waitTime=1
+kubeConfigFile=/tmp/byoh-cluster-kubeconfig
+for((i=1;i<=${maxRunTimes};i++));  
+do   
+    kubectl get secret/byoh-cluster-kubeconfig 2>&1 | grep -q "not found"
+    if [ $? -eq 0 ]; then
+        sleep ${waitTime}
+        continue
+    else 
+        kubectl get secret/byoh-cluster-kubeconfig -o json | jq -r .data.value | base64 --decode  > ${kubeConfigFile}
+        echo "Retrieve the kubeconfig of workload cluster successfully"
+        break
+    fi
+done
+
+
+echo "Applying a CNI for network..."
```

Comment:
We should allow for users to apply the CNI of their choice. I think it is better, we add a message that your cluster is provisioned, next step is to apply a CNI

Reply:
Again, I agree this being a getting-started script. But applying a CNI is out of scope of BYOH. What we claim is supporting different CNI, so our getting started script should not be biased to a particular CNI. 
Either give option on what cni user can install, or don't apply the CNI in the script - just print a message that byoh workload cluster is successfully created, next step is to apply a CNI of your choice.

Reply:
@anusha94 Installing CNI is also part of the content of getting_started.md. This script is just translate it to automatic script. And Installing CNI can also helps us to know better if our worker cluster is ok.

Reply:
Yes, so this script also should be documenting in the Getting Started guide.
If a user wants to use the script, they can do that or do the steps manually.
CNI is a part of getting started guide, because it makes the cluster "complete". But it is after byoh has executed without any error, we apply the CNI.

The getting started script should provide a similar experience IMO. Or take an input from user for CNI of their choice, and then calico can be the default option.